### PR TITLE
Edit user

### PR DESCRIPTION
**Pull request by ivanfelfo**
Feature: update users

### CODE REVIEW

File: app/controllers/v1/targets_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module V1
   class TargetsController < ApplicationController
     before_action :authenticate_v1_user!
+    after_action { pagy_headers_merge(@pagy) if @pagy }
```
Comment (martinjaimem) [inline app/controllers/v1/targets_controller.rb:4]:
```suggestion
```

File: spec/requests/users/user_update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        subject
+        expect(response).to have_http_status(:bad_request)
+      end
+    end
+  end
+
+  context 'when the user is not logged in' do
+    it 'responds with unauthorized' do
+      subject
+      v1_users_path(0)
```
Comment (martinjaimem) [inline spec/requests/users/user_update_spec.rb:51]:
```suggestion
```

File: spec/requests/users/user_update_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        let(:params) { { user: { gender: 'x' } } }
+
+        it 'returns code bad_request' do
+          subject
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it 'doesn\'t change user gender' do
+          subject
+          expect(user.reload.gender).not_to eq('x')
```
Comment (martinjaimem) [inline spec/requests/users/user_update_spec.rb:33]: expect { subject }.not_to change(user, :gender)

File: app/controllers/v1/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        @user = current_v1_user
+      else
+        render json: { error: 'Can\'t update other user attributes' }, status: :bad_request
+      end
+    end
+
+    private
+
+    def account_update_data
+      params.require(:user).permit(:gender, :email)
```
Comment (martinjaimem) [inline app/controllers/v1/users_controller.rb:21]: are we testing changing the email?

File: spec/requests/users/user_update_spec.rb (hunk 2 above)
Comment (ivanfelfo) [reply spec/requests/users/user_update_spec.rb:51]: 👍

File: app/controllers/v1/users_controller.rb (hunk 4 above)
Comment (ivanfelfo) [reply app/controllers/v1/users_controller.rb:21]: I wasn't able to update email

**Review by martinjaimem — APPROVED** [review]

**Review by juanmferreira93 — APPROVED** [review]

**Pull request merged**